ariadne/cardano/test/backend/Test/Spec/AcidState.hs

```diff
@@ -0,0 +1,97 @@
+module Test.Spec.AcidState (spec) where
+
+import Control.Exception.Base (ErrorCall(..), handle)
+import Control.Monad.Component hiding (throwM)
+
+import System.IO.Temp
+
+import Data.Acid (closeAcidState, openLocalStateFrom, query)
+import Data.Acid.Local (getState)
+
+import qualified Data.ByteString.Lazy as BL
+import Data.Text.Buildable (build)
+import Data.Text.Lazy.Builder (toLazyText)
+
+import Test.Hspec (Spec, describe)
+import Test.Hspec.QuickCheck (prop)
+import Test.QuickCheck (arbitrary, withMaxSuccess)
+import Test.QuickCheck.Monadic (monadicIO, pick, run)
+
+import Test.Spec.CreateWallet (NewWallet, applyNewWallet, genNewWalletRq)
+import Test.Spec.Fixture (genSpendingPasswords, withLayerLocalStorage)
+
+import Ariadne.Wallet.Cardano.Kernel.DB.AcidState (DB(..), Snapshot(..), defDB)
+import Ariadne.Wallet.Cardano.Kernel.DB.HdWallet (HdWallets(..))
+import Ariadne.Wallet.Cardano.Kernel.DB.Util.IxSet (IxSet, size)
+import Ariadne.Wallet.Cardano.Kernel.Internal (PassiveWallet(..), wallets)
+import Ariadne.Wallet.Cardano.WalletLayer (PassiveWalletLayer, pwlCreateWallet)
+
+spec :: Spec
+spec = describe "AcidState" $ do
+  prop "can open empty database" $ withMaxSuccess 1 $
+    monadicIO $ run $ withSystemTempDirectory "testWalletDBEmpty" $ \path -> do
+        getState path defDB 0 True >>= \case
+          Left _ -> return False
+          Right (acidDB, eventTag) -> do
+            db <- liftIO $ query acidDB Snapshot
+            let isEmptyDB = checkEmptyWalletDB db
+            return $ isEmptyDB && (eventTag == noEventsMsg)
+  prop "opened state is the same as it was before closing" $ withMaxSuccess 10 $ do

```

Comment:
This test is not a regression test for the bug and it wasn't requested in this issue. Please don't do more work than necessary.
Of course I am not asking to delete it, because it's already written and seems to work. This comment applies to further work.

Reply:
Ok. I know that the first test is enough to detect this specific bug, I thought that it would be good to detect both manifestations of the bug. It can help in future ). 

Anyway:

Reply:
This bug is a showstopper or at least critical, it should be fixed ASAP. If the regression test was hard to write I would even allow you to merge this PR without addition of any tests, but in this particular case the regression test is quite simple, that's why I want it to be done together with the fix (but not other more complex tests).